File: app/src/main/java/me/imzack/app/end/injector/component/AboutComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.AboutPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.AboutActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = AboutPresenterModule.class)
public interface AboutComponent {

    void inject(AboutActivity aboutActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/AboutComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.AboutPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.AboutActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(AboutPresenterModule::class))
interface AboutComponent {

    fun inject(aboutActivity: AboutActivity)
}
